openstf/model/objective.py

```diff
@@ -78,49 +92,135 @@ def __call__(self, trial: optuna.trial.FrozenTrial) -> float:
+        default_params = super().get_default_params(trial)
+
+        # Compare the list to the default parameter space
+        model_parameters = self.model.get_params()
+        keys = [x for x in model_parameters.keys() if x in default_params.keys()]
+        # create a dictionary with the matching parameters
+        model_params = {parameter: default_params[parameter] for parameter in keys}
+
+        # XGB specific parameters
         params = {
-            "eta": trial.suggest_float("eta", 0.01, 0.2),
-            "subsample": trial.suggest_float("subsample", 0.5, 1.0),
-            "min_child_weight": trial.suggest_int("min_child_weight", 1, 6),
-            "max_depth": trial.suggest_int("max_depth", 3, 10),
-            "gamma": trial.suggest_float("gamma", 0.0, 1.0),
-            "colsample_bytree": trial.suggest_float("colsample_bytree", 0.5, 1.0),
+            "gamma": trial.suggest_float("gamma", 1e-8, 1.0),
+            "booster": trial.suggest_categorical("booster", ["gbtree", "dart"]),
         }
-        return params
+        return {**model_params, **params}
+
+    def get_pruning_callback(self, trial: optuna.trial.FrozenTrial):
+        return optuna.integration.XGBoostPruningCallback(
+            trial, observation_key=f"validation_1-{self.eval_metric}"
+        )
+
+
+class LGBRegressorObjective(RegressorObjective):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.model_type = MLModelType.LGB
+
+    def get_params(self, trial: optuna.trial.FrozenTrial) -> dict:
+        default_params = super().get_default_params(trial)
+
+        # Compare the list to the default parameter space
+        model_parameters = self.model.get_params()
+        keys = [x for x in model_parameters.keys() if x in default_params.keys()]
+        # create a dictionary with the matching parameters
+        model_params = {parameter: default_params[parameter] for parameter in keys}
+
+        # LGB specific parameters
+        params = {
+            "num_leaves": trial.suggest_int("num_leaves", 16, 62),
+            "boosting_type": trial.suggest_categorical(
+                "boosting_type", ["gbdt", "dart", "rf"]
+            ),
+            "tree_learner": trial.suggest_categorical(
+                "tree_learner", ["serial", "feature", "data", "voting"]
+            ),
+            "n_estimators": trial.suggest_int("n_estimators", 50, 150),
+            "min_split_gain": trial.suggest_float("min_split_gain", 1e-8, 1),
+            "subsample_freq": trial.suggest_int("subsample_freq", 1, 10),
+        }
+        return {**model_params, **params}
+
+    def get_pruning_callback(self, trial: optuna.trial.FrozenTrial):
+        metric = self.eval_metric
+        if metric == "mae":
+            metric = "l1"
+        return optuna.integration.LightGBMPruningCallback(
+            trial, metric=metric, valid_name="valid_1"
+        )
+
+
+class XGBQRegressorObjective(RegressorObjective):
```

Comment:
I think we should use `XGBQuantileRegressorObjective` here. The one letter difference (compared to `XGBRegressorObjective`) is a little harder to see we use `XGBQuantileOpenstfRegressor` for the model as well.